### DISCUSSION TITLE
How to handle OS in combination with my Application?

### DISCUSSION THREAD

**Discussion by Jurrie** [DependencyTrack/dependency-track#1234] [Q&A]
I am currently looking into Dependency Track. I see I can define various [types of components](https://cyclonedx.org/use-cases/), such as 'Application' or 'Operating System'.

I don't fully understand how to use that for an application across various machines in various landscapes. For example, for my application I have a database machine, a machine running my application server (in which I deploy my application), and there are several other machines (an SFTP server, a firewall, maybe something like an LDAP machine, etc.) This landscape exists for "test", for "production" and maybe for other stages as well.

I understand that I can create a BOM [for my application](https://github.com/CycloneDX/cyclonedx-maven-plugin). I understand that I can create a BOM for each [machine's OS](https://github.com/CycloneDX/cyclonedx-linux-generator). But how do I structure these in Dependency Track?

Should I create one massive BOM? That would not reflect the actual situation: my application is updated out of sync with OS updates.
Should I create one project per machine, and one for my application? How can I then keep track which version of my application is deployed on which machine?

I feel I'm missing something obvious here 😀

**Comment by freddiN** [comment]
We too have the same issue, we cant really figure out how to use classifiers like "Firmware", "Container", "Operating System", "File" an so on: Do we still need to create a BOM for those?
We naively were hoping that we just could create e.g. "Debian 7" or "Glassfish 4.1", figure out the PURL, CPE and SWID and get them analyzed. 
The changelog for 4.0 states: "Added support for firmware and container component types", but at least to us the usage is unclear and also hard to figure out with information from the web, any shed of light here would be appreciated.

Regards
Freddi

**Reply by freddiN** [reply]
I think I figured it out:
When I create a project, the CPEs are not taken into account. But when I add a component to that project, the CVEs connected to that components CPE are correctly being shown.
